### PR TITLE
ARROW-7688: [JAVA] Update Checkstyle

### PR DESCRIPTION
**Pull request by Fokko**
Update Checkstyle to the latest version

### CODE REVIEW

**Comment by emkornfield** [comment]
I'm -0.5 on the bracket changes.  Can we preserve empty blocks on a single line?

File: java/dev/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -218,14 +219,8 @@
         <module name="JavadocMethod">
             <property name="scope" value="public"/>
             <property name="allowMissingParamTags" value="true"/>
-            <property name="allowMissingThrowsTags" value="true"/>
```
Comment (BryanCutler) [inline java/dev/checkstyle/checkstyle.xml]: Can you explain the removal of these rules? Are they no longer available and is there an alternative?

**Review by BryanCutler — CHANGES_REQUESTED** [review]
I'm also against the changes with brackets on the same line, could you add an exception that would allow the following?
```
private Constants() {}
```

**Comment by andygrove** [comment]
This PR seems to be stale and feedback has not been addressed so I am closing this for now.

**Pull request closed without merging**